src/sessions/video-session-handler.ts

```diff
@@ -216,33 +218,58 @@ export class VideoSessionHandler extends BaseSessionHandler {
   }
 
   // triggers a propose from the backend
-  async startSession (startParams: IStartVideoSessionParams): Promise<{ conversationId: string }> {
-    let participant: { address: string };
+  async startSession(startParams: IStartVideoSessionParams | IStartVideoMeetingSessionParams): Promise<{ conversationId: string }> {
+    if ((<IStartVideoSessionParams>startParams).jid) {
```

Comment:
If there's a better or preferred way to distinguish starting a video conference with a meeting id versus a jid, feel free to change this. At one point I had separated it into its own function and then realized that wouldn't work since SessionManager calls `startSession` for every handler and that didn't seem like the right place to distinguish between these. 😄

Reply:
So you can use type guards which is some typescript magic. You'd do something like this:
```
isVideoSessionParams (startParams: IStartVideoSessionParams | IStartVideoMeetingSessionParams): startParams is IStartVideoSessionParams {
    return !!(startParams as IStartVideoSessionParams).jid;
}

...

async startSession(startParams: IStartVideoSessionParams | IStartVideoMeetingSessionParams): Promise<{ conversationId: string }> {
    if (isVideoSessionParams(startParams)) {
        // typescript will now know that all references to `startParams` in this block are of type `IStartVideoSessionParams`
        return startVideoSession(startParams);
    } else if (isVideoMeetingParams(startParams)) {
        // typescript will know startParams is IStartVideoMeetingSessionParams
        return startVideoMeetingSession(startParams);
    }
}

private async startVideoSession (startParams: IstartVideoSessionParams) ... {
    // same stuff as we used to have
}
```
I think it also makes sense to turn `startSession` into a dispatcher session, then have private functions that perform the specific actions we want.

Reply:
If I'm reading this right, there's another variation of type guard using the `in` operator: https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-the-in-operator

Do you know of any technical reason to prefer one or the other? If not, am I correct in understanding you prefer the variation that defines a function with a type predicate (your version)?